PyStemmusScope/docker_process.py

```diff
@@ -0,0 +1,112 @@
+"""The Docker STEMMUS_SCOPE model process wrapper."""
+from PyStemmusScope.config_io import read_config
+from pathlib import Path
+import os
+import docker
+
+
+def make_docker_vols_binds(cfg_file: str) -> tuple[list[str], list[str]]:
+    """Make docker volume mounting configs.
+
+    Args:
+        cfg_file: Location of the config file
+
+    Returns:
+        volumes, binds
+    """
+    cfg = read_config(cfg_file)
+    
+    volumes = [cfg["OutputPath"], cfg["InputPath"]]
+    binds = [
+        f"{cfg['OutputPath']}:{cfg['OutputPath']}:rw",
+        f"{cfg['InputPath']}:{cfg['InputPath']}:ro",
+    ]
+
+    if (
+        not Path(cfg_file).parent.is_relative_to(cfg["InputPath"]) or
+        not Path(cfg_file).parent.is_relative_to(cfg["OutputPath"])
+    ):
+        cfg_folder = str(Path(cfg_file).parent)
+        volumes.append(cfg_folder)
+        binds.append(f"{cfg_folder}:{cfg_folder}:ro")
+
+    return volumes, binds
+
+
+class StemmusScopeDocker:
+    """Communicate with a STEMMUS_SCOPE Docker container."""
+    # The image is hard coded here to ensure compatiblity:
+    image = "ghcr.io/ecoextreml/stemmus_scope:1.5.0"
+
+    _process_ready_phrase = b"Select BMI mode:"
+
+    def __init__(self, cfg_file: str):
+        """Create the Docker container.."""
+        self.cfg_file = cfg_file
+        
+        self.client = docker.APIClient()
```

Comment:
I found that the docker run the model as `root` user. It changes the permissions of the `Output` directory on the host to `root:root`. Therefore,  `h5f.open` returns `PermissionError`. It should be a way to pass the argument `--user $(id -u)` that is often used in docker run command to APIClient. I amnot sure if the parameter `user` in [`create_container`](https://docker-py.readthedocs.io/en/stable/api.html?highlight=create_container#docker.api.container.ContainerApiMixin.create_container) does the job or parameter `mode` in [`ConfigReference`](https://docker-py.readthedocs.io/en/stable/api.html?highlight=ConfigReference#docker.types.ConfigReference).

Reply:
Thanks for figuring this out. Strangely I did not encounter this issue at all, it works completely fine on my system.

The documentation of the Docker SDK is not very good, so it can be very difficult to figure out what to do.

Reply:
Setting the `user` in `create_container` breaks things, because then the container has no access to its files (e.g matlab runtime or STEMMUS_SCOPE). This _could_ be fixable in the docker image but doesn't seem ideal...

Reply:
I have fixed this issue. It required a few steps. The `user=os.getuid()` kwarg is set when creating the container.
Additionally, the docker image had to be modified so it could run with a non-root user.

Lastly, this caused issues with the volume binds, which I fixed as well.